src/Symfony/Component/Security/Acl/Domain/ObjectIdentity.php

```diff
@@ -66,7 +66,7 @@ public static function fromDomainObject($domainObject)
             if ($domainObject instanceof DomainObjectInterface) {
                 return new self($domainObject->getObjectIdentifier(), ClassUtils::getRealClass($domainObject));
```

Comment:
I don't think so. In this case, returning a string is mandatory if you want to properly implement the `DomainObjectInterface`. We don't do type checks for scalars anywhere else. Do you think we should introduce that here?